modules/xmpp/xmpp.js

```diff
@@ -553,4 +580,91 @@ export default class XMPP extends Listenable {
 
         return details;
     }
+
+    /**
+     * Notifies speaker stats component if available that we are the new
+     * dominant speaker in the conference.
+     * @param {String} roomJid - The room jid where the speaker event occurred.
+     */
+    sendDominantSpeakerEvent(roomJid) {
+        // no speaker stats component advertised
+        if (!this.speakerStatsComponentAddress || !roomJid) {
```

Comment:
I'm wondering if this is room specific function why it's not defined in ChatRoom.js?

Reply:
Because of the speakerStatsComponentAddress which is retrieved here in xmpp.js and this is send from statistics.js where there is no room ... it can be handled in JitsiConferenceEventManager and can be made per room and in xmpp there can be a getter for speakerStatsComponentAddress if you think that is better.